tools/Customization/DevHome.Customization/ViewModels/DevDriveInsightsViewModel.cs

```diff
@@ -227,33 +232,56 @@ public void UpdateListViewModelList()
         DevDriveLoadingCompleted = true;
     }
 
-    private string? GetExistingCacheLocation(string rootDirectory, string targetDirectoryName)
-    {
-        var fullDirectoryPath = rootDirectory + targetDirectoryName;
-        if (Directory.Exists(fullDirectoryPath))
+    private readonly List<DevDriveCacheData> _cacheInfo =
+    [
+        new DevDriveCacheData
+        {
+            CacheName = "Pip cache (Python)",
+            EnvironmentVariable = "PIP_CACHE_DIR",
+            CacheDirectory = new List<string>
+            {
+                Path.Join(_localAppDataPath, "pip", "cache"),
+                Path.Join(_localAppDataPath, "packages", "PythonSoftwareFoundation.Python"),
+            },
+            ExampleDirectory = Path.Join("D:", "packages", "pip", "cache"),
+        },
+        new DevDriveCacheData
         {
-            return fullDirectoryPath;
+            CacheName = "NuGet cache (dotnet)",
+            EnvironmentVariable = "NUGET_PACKAGES",
+            CacheDirectory = new List<string> { Path.Join(_userProfilePath, ".nuget", "packages") },
+            ExampleDirectory = Path.Join("D:", "packages", "NuGet", "Cache"),
         }
-        else
+    ];
+
+    private string? GetExistingCacheLocation(DevDriveCacheData cache)
+    {
+        foreach (var cacheDirectory in cache.CacheDirectory!)
         {
-            var subDirPrefix = rootDirectory + "\\Packages\\PythonSoftwareFoundation.Python";
-            var subDirectories = Directory.GetDirectories(rootDirectory + "\\Packages", "*", SearchOption.TopDirectoryOnly);
-            var matchingSubdirectory = subDirectories.FirstOrDefault(subdir => subdir.StartsWith(subDirPrefix, StringComparison.OrdinalIgnoreCase));
-            var alternateFullDirectoryPath = matchingSubdirectory + "\\localcache\\local" + targetDirectoryName;
-            if (Directory.Exists(alternateFullDirectoryPath))
+            if (Directory.Exists(cacheDirectory))
             {
-                return alternateFullDirectoryPath;
+                return cacheDirectory;
+            }
+            else
+            {
+                var subDirectories = Directory.GetDirectories(_localAppDataPath + "\\Packages", "*", SearchOption.TopDirectoryOnly);
+                var matchingSubdirectory = subDirectories.FirstOrDefault(subdir => subdir.StartsWith(cacheDirectory, StringComparison.OrdinalIgnoreCase));
+                var alternateFullDirectoryPath = matchingSubdirectory + "\\localcache\\local" + cacheDirectory;

```

Comment:
>var alternateFullDirectoryPath = matchingSubdirectory + "\\localcache\\local" + cacheDirectory; [](http://example.com/codeflow?start=16&length=95)

alternateFullDirectoryPath is incorrect. Did you test for pip cache?

Reply:
I tested. Worked for me. I will update

Reply:
Pushed :)

Reply:
> > var alternateFullDirectoryPath = matchingSubdirectory + "\localcache\local" + cacheDirectory;
> 
> alternateFullDirectoryPath is incorrect. Did you test for pip cache?
> 
> cacheDirectory is already Path.Join(_localAppDataPath, "packages", "PythonSoftwareFoundation.Python"), which is the entire path, why join with matchingSubdirectory again?
> 
> Plz test before pushing out iterations.

I tested before push and it was working, that is how it slipped through the cracks. I tested both pip caches and I removed my environment variable in testing